resqpy/organize.py

```diff
@@ -408,148 +365,85 @@ def create_xml(self, add_as_part = True, add_relationships = True, originator =
 
    def is_equivalent(self, other, check_extra_metadata = True):
       """Returns True if this feature is essentially the same as the other; otherwise False."""
-
       if other is None or not isinstance(other, TectonicBoundaryFeature): return False
       if self is other or bu.matching_uuids(self.uuid, other.uuid): return True
       if check_extra_metadata and not equivalent_extra_metadata(self, other): return False
       return self.feature_name == other.feature_name and self.kind == other.kind
 
-
-   def __eq__(self, other):
-      return self.is_equivalent(other)
-
-
-   def __ne__(self, other):
-      return not self.is_equivalent(other)
-
-
-   def create_xml(self, add_as_part = True, originator = None):
+   def create_xml(self, add_as_part = True, originator = None, reuse = True):
       """Creates a tectonic boundary feature organisational xml node from this tectonic boundary feature object."""
 
-      tbf = self.model.new_obj_node('TectonicBoundaryFeature')
-
-      if self.uuid is None:
-         self.uuid = bu.uuid_from_string(tbf.attrib['uuid'])
-      else:
-         tbf.attrib['uuid'] = str(self.uuid)
-
-      self.model.create_citation(root = tbf, title = self.feature_name, originator = originator)
+      if reuse and self.try_reuse(): return self.node  # check for reusable (equivalent) object
+      # create node with citation block
+      tbf = super().create_xml(add_as_part=False, originator=originator)
 
+      assert self.kind in self.valid_kinds
       kind_node = rqet.SubElement(tbf, ns['resqml2'] + 'TectonicBoundaryKind')
       kind_node.set(ns['xsi'] + 'type', ns['resqml2'] + 'TectonicBoundaryKind')
       kind_node.text = self.kind
 
       if add_as_part:
          self.model.add_part('obj_TectonicBoundaryFeature', self.uuid, tbf)
 
-      self.root_node = tbf
-
       return tbf
 
 
 
-class GeneticBoundaryFeature():
+class GeneticBoundaryFeature(BaseResqpy):
    """Class for RESQML Genetic Boundary Feature (horizon) organizational objects."""
 
-   def __init__(self, parent_model, root_node = None, extract_from_xml = True, kind = None, feature_name = None):
-      """Initialises a genetic boundary feature (horizon or geobody boundary) organisational object."""
-
-      if kind is None:
-         if not extract_from_xml: kind = 'horizon'
-      else:
-         assert kind in ['horizon', 'geobody boundary'], 'genetic boundary feature kind not recognized'
+   resqml_type = "GeneticBoundaryFeature"
+   feature_name = _alias_for_attribute("title")
+   valid_kinds = ('horizon', 'geobody boundary')
 
-      self.model = parent_model
-      self.root_node = root_node
-      self.uuid = None
+   def __init__(self, parent_model, root_node = None, uuid = None, kind = None, feature_name = None):
+      """Initialises a genetic boundary feature (horizon or geobody boundary) organisational object."""
       self.kind = kind
       self.absolute_age = None   # (timestamp, year offset) pair, or None
```

Comment:
Should the `age` attribute be settable from the init method?

Reply:
I believe this is a geological age, though a little strange that it is stored as a timestamp, year offset pair – that is probably RESQML being very rigorous about how it handles a point in time!

Reply:
And yes, I guess we should support it, even though no dataset has yet made use of it.

Reply:
I suggest we note this as a todo and leave it as is for now.